### PR TITLE
Implement basic guitar strumming

### PR DESCRIPTION
**Pull request by laurynaslubys**
Adresses #3. Implemented basic guitar strumming, including using guitar like chords. The diff is bigger than it should be, but I accidentally formatted the main file and noticed it too late. 

[Here's a sample](https://drive.google.com/file/d/1WfvLWiJcAxKEr6zPVze85ZdSeIF1xkFJ/view) of playing this through the free "[Ample Guitar M Lite](https://www.amplesound.net/en/download.asp)" plugin. 

Also I extended the harmony to expose the "flavour" (i.e. major, minor, etc) of each chord in a scale. There's got to be a better term for this, but I couldn't find it.

### CODE REVIEW

**Comment by laurynaslubys** [comment]
Implement a bunch more strumming patterns, they can be switched using midi notes 67-72 for demo purposes.

**Comment by laurynaslubys** [comment]
This doesn't take into consideration tension and energy levels, as I don't really understand the concept at the moment.

**Comment by emilis-wix** [comment]
I'm not a guitar player and to me it sounds very natural.
Amazing job!

Tested it on my Yamaha MODX guitar category.

There are some glitches though: after some time MODX stops playing.
I'm guessing it doesn't receive enough note-offs or something.

It is also hard to time the chord key and the strum key on the keyboard. It seems to me that pressing a chord key cuts off the playing notes. Should it be like this, or should we only cut off what's playing on Up/Down?

File: midi-magic/src/guitar/strumming.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  };
+  const strum = (notes: number[]) => {
+    notesOut.value?.sendAllNotesOff();
+    const varietyNumber = Math.random();
+    const baseVelocity = calcVelocity(varietyNumber, opts);
+    notes.forEach((chordNote, index) => {
+      notesOut.value?.sendNoteOn(
+        new Note(chordNote, {
+          attack: baseVelocity - opts.velocityDecay * index,
+        }),
```
Comment (emilis-wix) [inline midi-magic/src/guitar/strumming.ts:63]: Should we add a `duration: ...` here and below?

I noticed that the chord notes continue playing infinitely if not stopped by a chord change.

The duration could be quite long if seems necessary.

**Comment by emilis-wix** [comment]
I think the "my MODX stopped playing" bug happens in the FullStrumming mode.

We were playing a CombinedStrumming for some time and it worked OK. Then we switched to FullStrumming and my MODX stopped playing after a short time.

File: midi-magic/src/guitar/strumming.ts (hunk 1 above)
Comment (laurynaslubys) [reply midi-magic/src/guitar/strumming.ts:63]: My approach was to model a real guitar: after you strum a chord, it doesn't _really_ stop ringing until you release your left hand. We could either try to add some long duration here or actually keep track of ringing notes and send note offs explicitly. I'm leaning more on keeping track of currently ringing notes.

**Comment by laurynaslubys** [comment]
> I'm not a guitar player and to me it sounds very natural. Amazing job!
> 
> Tested it on my Yamaha MODX guitar category.
> 
> There are some glitches though: after some time MODX stops playing. I'm guessing it doesn't receive enough note-offs or something.

You're probably right, I was hoping to get away with sending all notes off, but it's possible some devices don't play well with that. 
 
> It is also hard to time the chord key and the strum key on the keyboard. It seems to me that pressing a chord key cuts off the playing notes. Should it be like this, or should we only cut off what's playing on Up/Down?

That's kind of what happens on a guitar. We should test how if feels on the controller, then we can decide what we should do. But yes, you do need to time it right. To me it feels ok, when you're the one switching chords, not sure how it would feel in follower mode.

I suggest we merge this and I'll update #3 with the comments and concerns from here.

**Pull request merged**